File: src/olcw78 (die yoong)/jsAlgorithmStudy/week01/02_배열_제어하기.js
/**
 *
 * @param arr {number[]}
 */
function solution(arr) {
    const unq = [...new Set(arr)]
    unq.sort((a, b) => b - a);
    return unq
}

const r1 = solution([4, 2, 2, 1, 3, 4]);
console.log(r1)

const r2 = solution([2, 1, 1, 3, 2, 5, 4]);
console.log(r2)

File: src/olcw78 (die yoong)/jsAlgorithmStudy/week01/02_배열_제어하기.go
//package main
//
//import (
//	"fmt"
//	"slices"
//)
//
//func solution(arr []int) []int {
//	unq := make(map[int]int)
//	for _, e := range arr {
//		unq[e] = e
//	}
//
//	answer := make([]int, 0, len(unq))
//	for _, v := range unq {
//		answer = append(answer, v)
//	}
//
//	slices.SortFunc(answer,
//		func(a, b int) int {
//			return b - a
//		},
//	)
//
//	return answer
//}
//
//func main() {
//	r1 := solution([]int{4, 2, 2, 1, 3, 4})
//	fmt.Println(r1)
//
//	r2 := solution([]int{2, 1, 1, 3, 2, 5, 4})
//	fmt.Println(r2)
//}
